insights/parsers/softnet_stat.py

```diff
@@ -0,0 +1,106 @@
+number of packet drops packet_drops (second column), time squeeze i.e net_rx_action
+performed time_squeeze(third column), cpu collision i.e collision occur while obtaining
+device lock while transmitting cpu_collision packets (Ninth column), received_rps
+number of times cpu woken up received_rps (tenth column), number of times reached flow
+limit count flow_limit_count (eleventh column).
+
+    The typical contents of ``cat /proc/net/softnet_stat`` file is as below::
+
+        00008e78 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
+        000040ee 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
+        0001608c 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
+        0000372f 00000001 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
+
+        Column-01: packet_process: Packet processed by each CPU.
+        Column-02: packet_drop: Packets dropped.
+        Column-03: time_squeeze: net_rx_action.
+        Column-08: cpu_collision: collision occur while obtaining device lock while transmitting.
+        Column-09: received_rps: number of times cpu woken up received_rps.
+        Column-10: flow_limit_count: number of times reached flow limit count.
+
+    .. Note::
+                There is very less doc about the fields in the file, columns are not labeled and
+                could change between the kernel releases. This format is compatible for RHEL-6
+                and RHEL-7 releases. Also it is unlikely that the positions of those values will
+                change in short term.
+
+    Examples:
+        >>> SOFTNET_STATS = '''
+        ... 00008e78 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
+        ... 000040ee 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
+        ... 0001608c 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
+        ... 0000372f 00000001 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
+        ...'''.strip()
+        >>> softnet_obj = SoftNetStats(context_wrap(SOFTNET_STATS))
+        >>> softnet_obj.is_packet_drops()
+        True
+        >>> softnet_obj.cpu_instances
+        4
+        >>> softnet_obj.per_cpu_nstat('packet_drops')
+        [0, 0, 0, 1]
+"""
+
+from .. import Parser, parser
+
+
+@parser("softnet_stat")
+class SoftNetStats(Parser):
+    """Parses ``/proc/net/softnet_stat`` file contains"""
+
+    def __init__(self, context):
+
+        """ Total number of CPUs listed in the stats records"""
+        self.cpu_instances = 0
+
+        """ List of network stats per cpu instace """
+        self.cpu_nstats = []
+        super(SoftNetStats, self).__init__(context)
+
+    def parse_content(self, contents):
+        for line in contents:
+            stats = line.split(None)
```

Comment:
Don't need this line if you're then using the list comprehension to compile the stats line :-)